### PR TITLE
CycloneDX cleanups

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: reporter/src/funTest/kotlin/reporters/CycloneDxReporterFunTest.kt [hunk 1]
```diff
@@ -43,6 +42,12 @@ class CycloneDxReporterFunTest : WordSpec({
     val defaultSchemaVersion = CycloneDxReporter.DEFAULT_SCHEMA_VERSION.versionString
     val outputDir = createSpecTempDir()
 
+    fun String.patchCycloneDxResult() =
```
Comment (fviernau) [inline reporter/src/funTest/kotlin/reporters/CycloneDxReporterFunTest.kt:45]: The function does not use class context. 
I believe it's nicer to have such functions outside of the class, to not distract the reader with irrelevant implementation details.

**Review by fviernau — CHANGES_REQUESTED** [review]

File: reporter/src/funTest/kotlin/reporters/CycloneDxReporterFunTest.kt (hunk 1 above)
Comment (sschuberth) [reply reporter/src/funTest/kotlin/reporters/CycloneDxReporterFunTest.kt:45]: I'd usually agree, but here I chose to put the function next to the properties as it seemed nicer to me to have all "other stuff" (than tests) in a single place than spread over the file.

Comment (fviernau) [reply reporter/src/funTest/kotlin/reporters/CycloneDxReporterFunTest.kt:45]: > but here I chose to put the function next to the properties as it seemed nicer to me to have all "other stuff" (than tests) in a single place 

Hm, the first commit in this PR seems to move away from that "single place" closer to the place of use. So, looking at both commits together this argumentation seems to be a bit inconsistent.

Comment (sschuberth) [reply reporter/src/funTest/kotlin/reporters/CycloneDxReporterFunTest.kt]: Yeah, I can see what you mean. Ok, I've moved the function outside of the class.

**Review by fviernau — APPROVED** [review]

**Pull request merged**